### PR TITLE
Introduce RemoveOperation

### LINKED ISSUE #623: ManagedGroup API: Remove: cannot determine if someone left or was removed
We cannot determine if someone left or was removed (also: distinction between own member and other members)

### PR DESCRIPTION
**Pull request by raphaelrobert**
This PR introduces the `RemoveOperation` `enum` that helps interpret the semantic value of a remove proposal when inspecting proposals that are covered by a Commit.

Fixes #623.

### CODE REVIEW

File: openmls/src/group/tests/test_remove_operation.rs [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+        // === Remove operation ===
+
+        let alice_kpr = *alice_group
+            .key_package_ref()
+            .expect("An unexpected error occurred.");
+        let bob_kpr = *bob_group
+            .key_package_ref()
+            .expect("An unexpected error occurred.");
+
+        // We differentiate between th two test cases here
```
Comment (kkohbrok) [inline openmls/src/group/tests/test_remove_operation.rs:136]:
```suggestion
        // We differentiate between the two test cases here
```

File: openmls/src/group/tests/test_remove_operation.rs [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+        // === Remove operation from Alice's perspective ===
+
+        let alice_staged_commit = alice_group.pending_commit().expect("No pending commit.");
+
+        let remove_proposal = alice_staged_commit
+            .remove_proposals()
+            .next()
+            .expect("An unexpected error occurred.");
+
+        let remove_operation = RemoveOperation::new(remove_proposal, &alice_group)
```
Comment (kkohbrok) [inline openmls/src/group/tests/test_remove_operation.rs:182]: So the consumer inspecting the proposal has to create the `RemoveOperation` themselves? For inspection purposes, wouldn't it be more convenient for the consumer if they had all the information available immediately? Don't get me wrong, this is a great change already, I'm just wondering what we want the consumer API to look like eventually.

**Review by kkohbrok — COMMENTED** [review]
Thanks for the PR! Looks good to me already with just one typo and one question.

File: openmls/src/group/tests/test_remove_operation.rs (hunk 2 above)
Comment (raphaelrobert) [reply openmls/src/group/tests/test_remove_operation.rs:182]: As discussed, I think it's ok to just have it as an external helper for now. The current layout makes it easier to treat all remove proposals the same, which makes bulk parsing easier.
We could also add something similar for add and update proposals (although it's less needed there).

**Review by kkohbrok — APPROVED** [review]
Thanks!

**Pull request merged**